changes/2118.feature

```diff
@@ -0,0 +1 @@
+Refactor DynamicResource construction to facilitate subclassing from DynamicResource.
```

Comment:
Please drop the file: the change is not related to public API, I don't want to attract attention by this record